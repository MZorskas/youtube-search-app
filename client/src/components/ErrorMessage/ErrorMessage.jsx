import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import './ErrorMessage.scss';

const ErrorMessage = ({ message, className }) => {
  return (
    <p
      data-testid="error-message"
      className={classNames('error-message', className)}
    >
      {message}
    </p>
  );
};

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default ErrorMessage;
