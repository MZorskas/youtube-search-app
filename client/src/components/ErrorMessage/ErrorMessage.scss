@import 'src/styles/mixins/index';
@import 'src/styles/base/base';

.error-message {
  @include text-style-body;

  color: $color-font-primary;
  margin: 1.5rem auto;
}
