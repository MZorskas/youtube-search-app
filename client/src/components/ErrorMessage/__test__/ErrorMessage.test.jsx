import React from 'react';
import ReactDom from 'react-dom';
import ErrorMessage from '../ErrorMessage';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('ErrorMessage component', () => {
  it('renders with propTypes warning', () => {
    const spy = jest.spyOn(global.console, 'error');
    const div = document.createElement('div');
    ReactDom.render(<ErrorMessage />, div);
    expect(spy).toHaveBeenCalled();
  });

  it('renders without crashing', () => {
    const spy = jest.spyOn(global.console, 'error');
    const div = document.createElement('div');
    ReactDom.render(<ErrorMessage message="" />, div);
    expect(spy).not.toHaveBeenCalled();
  });

  it('renders with correct message', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<ErrorMessage message="Error message" />);
    expect(screen.getByTestId('error-message').textContent).toBe(
      'Error message'
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('renders with additional className', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <ErrorMessage className="error-message-test" message="Error message" />
    );

    expect(screen.getByTestId('error-message').className).toBe(
      'error-message error-message-test'
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(<ErrorMessage message="Error message" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with additional className', () => {
    const tree = renderer
      .create(
        <ErrorMessage message="Error message" className="error-message-test" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
