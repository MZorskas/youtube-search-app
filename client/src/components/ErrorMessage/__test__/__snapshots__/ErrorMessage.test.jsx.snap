// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component matches snapshot 1`] = `
<p
  className="error-message"
  data-testid="error-message"
>
  Error message
</p>
`;

exports[`ErrorMessage component matches snapshot with additional className 1`] = `
<p
  className="error-message error-message-test"
  data-testid="error-message"
>
  Error message
</p>
`;
